junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/AbstractTestDescriptor.java

```diff
@@ -147,6 +150,22 @@ public void removeFromHierarchy() {
 		this.children.clear();
 	}
 
+	@Override
+	public void orderChildren(UnaryOperator<List<TestDescriptor>> orderer) {
+		Preconditions.notNull(orderer, "orderer must not be null");
+		List<TestDescriptor> suggestedOrder = orderer.apply(new ArrayList<>(this.children));
+		Preconditions.notNull(suggestedOrder, "orderer may not return null");
+
+		Set<? extends TestDescriptor> orderedChildren = new LinkedHashSet<>(suggestedOrder);
+		boolean unmodified = this.children.equals(orderedChildren);
+		Preconditions.condition(unmodified, "orderer may not add or remove test descriptors");
+		Preconditions.condition(this.children.size() == suggestedOrder.size(),
+			"orderer may not add duplicate test descriptors");
+
+		this.children.clear();
+		this.children.addAll(orderedChildren);
```

Comment:
Here too there is the possibility here that `orderedChildren` are impersonating the actual `children`.

Reply:
I think we can live with that. It's clearly documented that an orderer must not do that. If anyone does, all bets are off. Besides, one can already call `removeChild` and `addChild` to achieve impersonation. :slightly_smiling_face: